sumo/electronic_structure/dos.py

```diff
@@ -125,7 +125,7 @@ def load_dos(
                     site, orbital
                 ).get_smeared_densities(gaussian)
 
-    if vr.parameters["LSORBIT"]:
+    if vr.parameters["LSORBIT"] and Spin.down in dos.densities:
```

Comment:
Alternatively, this should work:
```
    if vr.parameters["LSORBIT"]:
        # pymatgen includes the spin down channel for SOC calculations, even
        # though there is no density here. We remove this channel so the
        # plotting is easier later on.
        dos.densities.pop(Spin.down, None)
        for site in dos.pdos:
            for orbital in dos.pdos[site]:
                dos.pdos[site][orbital](Spin.down, None)
```

I don't have a very strong opinion on which is more readable.

Either way, would be good to update the comment as well to clarify "pre-2025 pymatgen". Then perhaps one day somebody will read it, notice that our minimum version has increased, and delete the unnecessary code 😄

Reply:
For fun/education, here is an alternative that avoids the nested loops and unnecessary key lookups:

```python
from itertools import chain
from operator import methodcaller
remove_spin_down = methodcaller("pop", Spin.down, None)

all_orbitals = chain(*map(methodcaller("values"), dos.pdos.values()))

remove_spin_down(dos.densities)
for orbital in all_orbitals:
    remove_spin_down(orbital)
```

But it's quite a different style from most of Sumo so this might not be the best place to start using it!

Reply:
Updated to the `pop` suggestion and with the `pymatgen<2025` comment